group-generators/helpers/data-providers/token-provider/index.ts

```diff
@@ -7,68 +7,101 @@ import {
 import { FetchedData } from "topics/group";
 
 export class TokenProvider {
-  public async getNftHolders({
+  public async getERC721Holders({
     contractAddress,
     network,
+    minAmount,
+    forcedValue,
+    snapshot,
   }: {
     contractAddress: string;
     network?: string;
+    minAmount?: number;
+    forcedValue?: number;
+    snapshot?: string;
   }): Promise<FetchedData> {
     const bigQueryProvider = new BigQueryProvider({
       network: fromStringToSupportedNetwork(network ?? SupportedNetwork.MAINNET),
     });
-    const rawData: FetchedData = await bigQueryProvider.getNftHolders({
+    const rawData: FetchedData = await bigQueryProvider.getERC721Holders({
       contractAddress,
+      snapshot,
     });
 
+    // Filter holders by minAmount
     const data: FetchedData = {};
     for (const key of Object.keys(rawData)) {
-      data[key] = 1;
+      const value = BigNumber.from(rawData[key]);
+      if(minAmount) {
+        const minAmountBig = BigNumber.from(minAmount);
+        if (value.gte(minAmountBig)) {
+          data[key] = forcedValue ?? BigNumber.from(rawData[key]).toString();
+        }
+      }
+      else {
+        data[key] = forcedValue ?? BigNumber.from(rawData[key]).toString();
+      }
     }
     return data;
   }
 
-  public async getNftHoldersCount({
+  public async getERC721HoldersCount({
     contractAddress,
     network,
   }: {
     contractAddress: string;
     network?: string;
   }): Promise<number> {
-    const bigQueryProvider = new BigQueryProvider({
-      network: fromStringToSupportedNetwork(network ?? SupportedNetwork.MAINNET),
-    });
-    return bigQueryProvider.getNftHoldersCount({
+    const data = await this.getERC721Holders({
```

Comment:
Why not use the ERC721HoldersCount function instead?

Reply:
Good catch, you're right! My bad

Reply:
Because ERC721HoldersCount function calculate the number of txs found and not the number of holders.
By using the same getERC721Holders function, we are sure that the Group Generator will work

Reply:
Good catch! I'm actually divided by taking the result of the getERC721Holders function rather than going through a getERC721HoldersCount. Because with getERC721Holders you're sure to get the same result when generating the group generator.
And in terms of consumption, in all cases we're obliged to recalculate the latests txs, so it's all the same (there was an error in my count functions in relation to this, I've corrected it)